### PR TITLE
Fixes MGDCTRS-686, fixes MGDCTRS-685, renamed feature flag to connector-enable-unassigned-connectors, added stringent checks in connector patch handler

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
Various fixes for minor issues. Renamed a feature flag to be more generic. Added stricter state transition checks in connector PATCH operation. Minor fix in connector tests.  

## Verification Steps
CI tests should pass without errors.

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [X] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [X] Documentation added for the feature
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has created for changes required on the client side~~

### CODE REVIEW

**Comment by dhirajsb** [comment]
@lburgazzoli @valdar please review and merge.

File: internal/connector/test/integration/features/connector-agent-api.feature [hunk 1]
```diff
@@ -7,8 +7,7 @@ Feature: connector agent API
   Background:
     Given the path prefix is "/api/connector_mgmt"
     Given an org admin user named "Jimmy"
-    Given I store orgid for "Jimmy" as ${jimmy_org_id}
-    Given an org admin user named "Bobby" in organization "${jimmy_org_id}"
+    Given an org admin user named "Bobby" in organization "${Jimmy.OrgId}"
```
Comment (valdar) [inline internal/connector/test/integration/features/connector-agent-api.feature:10]: This seems to be the only change in tests, is this sufficient to test all the changes?

**Review by valdar — CHANGES_REQUESTED** [review]

File: internal/connector/internal/handlers/connectors.go [hunk 2]
```diff
@@ -178,15 +188,39 @@ func (h ConnectorsHandler) Patch(w http.ResponseWriter, r *http.Request) {
...
 			// over the fields that they are allowed to modify..
 			resource.Name = patch.Name
 			resource.Connector = patch.Connector
-			resource.DesiredState = patch.DesiredState
 			resource.Kafka = patch.Kafka
 			resource.ServiceAccount = patch.ServiceAccount
 			resource.SchemaRegistry = patch.SchemaRegistry
-			resource.NamespaceId = patch.NamespaceId
+
+			if h.connectorsConfig.ConnectorEnableUnassignedConnectors {
```
[the remark this answers is not in the document]
Comment (dhirajsb) [reply internal/connector/internal/handlers/connectors.go:215]: yeah, it's kinda ignore right now. Can you create an issue and we can work on improving it later? Let's get the staging release done this week.

File: internal/connector/internal/handlers/connectors.go [hunk 3]
```diff
@@ -222,19 +256,20 @@ func (h ConnectorsHandler) Patch(w http.ResponseWriter, r *http.Request) {
 				return nil, svcErr
 			}
 
-			serr = h.connectorsService.Update(r.Context(), p)
-			if serr != nil {
-				return nil, serr
-			}
-
+			// update connector phase before desired state
```
[the remark this answers is not in the document]
Comment (dhirajsb) [reply internal/connector/internal/handlers/connectors.go:259]: Agent acts on changes it sees in desired state, and responds with updated deployment phase, which is copied into component status. So updating phase first makes sure agent doesn't see the desired state change request before that.

File: internal/connector/test/integration/features/connector-agent-api.feature (hunk 1 above)
Comment (dhirajsb) [reply internal/connector/test/integration/features/connector-agent-api.feature:10]: The changes are guards for bad data, which we don't have right now. We can add more negative tests in the future.

Comment (dhirajsb) [reply internal/connector/test/integration/features/connector-agent-api.feature:10]: See issue https://issues.redhat.com/browse/MGDCTRS-695 for future enhancements.

File: internal/connector/internal/handlers/connectors.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (valdar) [reply internal/connector/internal/handlers/connectors.go:259]: ah I see tnx.

**Review by valdar — APPROVED** [review]

File: internal/connector/internal/handlers/connectors.go [hunk 4]
```diff
@@ -88,7 +99,10 @@ func (h ConnectorsHandler) Create(w http.ResponseWriter, r *http.Request) {
 			convResource.Owner = user.UserId()
 			convResource.OrganisationId = user.OrgId()
 
-			// validate create operation if connector is assigned to a namespace
+			// namespace id is a required field if unassigned connectors are not supported
+			if !h.connectorsConfig.ConnectorEnableUnassignedConnectors && (convResource.NamespaceId == nil || *convResource.NamespaceId == "") {
```
Comment (lburgazzoli) [inline internal/connector/internal/handlers/connectors.go:103]: Wonder why we don't set `namespace_id` as required in the OpenAPI definition
 I don't think we have a case where a user can create a connect without having to set the `namespace_id`

**Pull request merged**